Detectors/FOCAL/simulation/src/Detector.cxx

```diff
@@ -165,6 +184,54 @@ Bool_t Detector::ProcessHits(FairVolume* v)
     }
   }
 
+  // Processing ECAL hits
+  if (TVirtualMC::GetMC()->CurrentMedium() == mMedSensECal) {
+    LOG(debug) << "We are in sensitive volume " << v->GetName() << ": " << TVirtualMC::GetMC()->CurrentVolPath();
+
+    double eloss = TVirtualMC::GetMC()->Edep() * 1e9; // energy in eV  (GeV->eV)
+    if (eloss < DBL_EPSILON) {
+      return false; // only process hits which actually deposit some energy in the FOCAL
+    }
```

Comment:
Could be potentially dangerous, as it might shadow other parts (i.e. pixel simulation which comes later). I think we need to outsource the functions (also for HCAL @hahassan7) into dedicated process functions.

Reply:
ProcessHit method is split into two parts. It contains two submethods for ECAL and HCAL, separatelly.